### PR TITLE
scripts: handle image format

### PR DESCRIPTION
**Pull request by arkivm**
Looking at file extension to determine the image format does not always work.  e.g., cloud-init images come with an .img extension but are qcow2 by default. This would result in a boot failure if we try to pass the format as "raw" instead of qcow2.

Instead, use `qemu-img` tool that determines the image format by actually looking into the file contents.

Signed-off-by: Vikram Narayanan <vikram186@gmail.com>

### CODE REVIEW

File: scripts/launch-qemu.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# SPDX-License-Identifier: MIT
 #!/bin/bash
+# SPDX-License-Identifier: MIT
```
Comment (Zildj1an) [inline scripts/launch-qemu.sh:2]: Why did you move this? Were you having issues with `#!/bin/bash` on the second line? If you were not facing error, let's not change this. If you were, send a separate patch changing the ordering in all four scripts. One way or the other, this should leave this commit.

File: scripts/launch-qemu.sh [hunk 2]
```diff
@@ -332,38 +332,51 @@ add_opts "-drive if=pflash,format=raw,unit=1,file=${UEFI_BIOS_VARS}"
 # add CDROM if specified
 [ -n "${CDROM_FILE}" ] && add_opts "-drive file=${CDROM_FILE},media=cdrom,index=0"
 
+get_image_format() {
+	IMAGE_FILE=$1
+	if [ ! -f "${IMAGE_FILE}" ]; then
+		echo "${LINENO} Error: Missing argument or incorrect paramter \"$1\" for function ${FUNCNAME}";
```
Comment (Zildj1an) [inline scripts/launch-qemu.sh:338]: No need to say `{LINENO} Error:` or `for function ${FUNCNAME}`.  We don't do it like that elsewhere, it may confuse users, and developers don't really need it. Also, fix typo in 'paramter'.

**Review by Zildj1an — CHANGES_REQUESTED** [review]
Thanks, this looks good to me :+1:. There's minor changes to be made before I can merge, I have added comments. You can amend and add my Reviewed-By tag. By the way, that tag should go after your signed off, look at [this](https://lkml.org/lkml/2022/11/17/946) example.

File: scripts/launch-qemu.sh (hunk 1 above)
Comment (arkivm) [reply scripts/launch-qemu.sh:2]: Because, that's how shebangs work. Else, this would be executed under whatever is pointed to by `/bin/sh` and not by `/bin/bash` as you would expect. How [kernel](https://github.com/torvalds/linux/blob/81ac25651a62c958bb0e074e0d4e25060ea557dd/fs/binfmt_script.c#L41) interprets it. For comparison, here is a how SPDX's are added to [kernel scripts](https://github.com/torvalds/linux/blob/master/kernel/gen_kheaders.sh#L2). [Github code search](https://github.com/search?utf8=%E2%9C%93&q=SPDX-License-Identifier+lang%3AShell+&type=code) of scripts containing SPDX.

Agree with moving it to a separate patch. will amend.

File: scripts/launch-qemu.sh (hunk 2 above)
Comment (arkivm) [reply scripts/launch-qemu.sh:338]: Sure. I was just giving more information on the error. I can remove it.

**Comment by arkivm** [comment]
@Zildj1an The [example](https://lkml.org/lkml/2022/11/17/946) has `Reviewed-by` after the `Signed-off-by`. But you wanted it the other way, right?

**Comment by Zildj1an** [comment]
> 



> @Zildj1an The [example](https://lkml.org/lkml/2022/11/17/946) has `Reviewed-by` after the `Signed-off-by`. But you wanted it the other way, right?

@arkivm well, that's bad luck... I just took a random patch and assumed they were following that order. I think that your Signed-off-by should go last, yes. [This](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/kernel?h=v6.1-rc5&id=e7fd8b68404f3d8bc03f85bc3c078d67096be06f) is a good example in the kernel. Or [this](https://github.com/AMDESE/linux-svsm/commit/3e61a1e3177258e119843de61aa2540e69c7cb2d) older commit to our project.

File: scripts/launch-qemu.sh (hunk 1 above)
Comment (Zildj1an) [reply scripts/launch-qemu.sh]: Yep, makes sense. Looking forward to that separate patch.
